staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go

```diff
@@ -61,16 +88,28 @@ func SchemaDeclType(s *schema.Structural, isResourceRoot bool) *DeclType {
 	case "array":
 		if s.Items != nil {
 			itemsType := SchemaDeclType(s.Items, s.Items.XEmbeddedResource)
+			var maxItems int64
+			if s.Items.ValueValidation != nil && s.Items.ValueValidation.MaxItems != nil {
+				maxItems = *s.Items.ValueValidation.MaxItems
```

Comment:
Something is wrong here. I was expecting:

```
if s.ValueValidation != nil && s.ValueValidation.MaxItems != nil {
  maxItems = *s.ValueValidation.MaxItems
```

For comparison see the below "object" case, which I is handling this right.

The reason tests are passing is that they make the same wrong assumption.  I've added a comment in the tests as well.

Reply:
Ah, you're right. I'll mark this as resolved once I fix this and the associated unit tests.